### ISSUE TITLE
Failure's __repr__ should show the value of the exception

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8004]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#8004|
|Type|enhancement|
|Created|2015-08-22 16:34:13Z|
|Branch|https://github.com/twisted/twisted/tree/failure-repr-8004|

Failure's ```__repr__``` currently looks something like this:

```<twisted.python.failure.Failure <type 'exceptions.NameError'>```

I posit that the following would be a lot better:

```<twisted.python.failure.Failure NameError: global name 'foo' is not defined>```

<details><summary>Searchable metadata</summary>

```
trac-id__8004 8004
type__enhancement enhancement
reporter__radix radix
priority__normal normal
milestone__ 
branch__branches_failure_repr_8004 branches/failure-repr-8004
branch_author__radix radix
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1440261253185677 1440261253185677
changetime__1440612460325197 1440612460325197
version__None None
owner__radix radix

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_1"></a>|@radix commented|
|-|-|

(In [45510]) Branching to failure-repr-8004.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_3"></a>|@radix removed owner|
|-|-|

Okay, I think this is ready.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set owner to @radix|
|-|-|

Thanks for working on this.

I think that this is a good improvement.

Please add a new newsfile... it can be a feature ... and them merge. :)


Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_5"></a>|@radix commented|
|-|-|

Thanks very much for the quick review adiroiban :)

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_6"></a>|@radix set status to `closed`|
|-|-|

(In [45559]) Merge failure-repr-8004: include exception message in Failure repr

Author: radix
Reviewer: adiroiban
Fixes: [#8004](https://github.com/twisted/twisted/issues/8004)

Failure.__repr__ now includes the exception message in addition
to the type.
